arctic/chunkstore/chunkstore.py

```diff
@@ -239,37 +242,43 @@ def read(self, symbol, chunk_range=None, filter_data=True, **kwargs):
         -------
         DataFrame or Series
         """
-
         sym = self._get_symbol_info(symbol)
         if not sym:
             raise NoDataFoundException('No data found for %s' % (symbol))
 
-        spec = {SYMBOL: symbol,
-               }
+        # special handling based on a single symbol or multi symbol read
+        spec = {SYMBOL: {'$in': symbol}} if isinstance(symbol, list) else {SYMBOL: symbol}
+        chunker = CHUNKER_MAP[sym[0][CHUNKER]] if isinstance(symbol, list) else CHUNKER_MAP[sym[CHUNKER]] 
+        deser = SER_MAP[sym[0][SERIALIZER]].deserialize if isinstance(symbol, list) else SER_MAP[sym[SERIALIZER]].deserialize 
+
 
         if chunk_range is not None:
-            spec.update(CHUNKER_MAP[sym[CHUNKER]].to_mongo(chunk_range))
+            spec.update(chunker.to_mongo(chunk_range))
 
-        by_start_segment = [(START, pymongo.ASCENDING),
+        by_start_segment = [(SYMBOL, pymongo.ASCENDING),
+                            (START, pymongo.ASCENDING),
                             (SEGMENT, pymongo.ASCENDING)]
         segment_cursor = self._collection.find(spec, sort=by_start_segment)
 
-        chunks = []
+        chunks = defaultdict(list)
         for _, segments in groupby(segment_cursor, key=lambda x: x[START]):
             segments = list(segments)
-            mdata = self._mdata.find_one({SYMBOL: symbol, START: segments[0][START], END: segments[0][END]})
+            mdata = self._mdata.find_one({SYMBOL: segments[0][SYMBOL],
```

Comment:
I think it would be a very useful feature. Especially in equities where the universe is very large but on a given day we trade a sub set of this universe. These leads to a tension between all equity data in a huge single symbol versus a symbol per equity. With an efficient multi symbol read the symbol per equity becomes more attractive.

I think the multiple symbols with the same start data can be solved with:
` for _, segments in groupby(segment_cursor, key=lambda x: (x[START], x[SYMBOL])):`
(untested!)